lib/rspec/mocks/matchers/have_received.rb

```diff
@@ -107,7 +107,11 @@ def generate_failure_message
 
         def expected_messages_received_in_order?
           mock_proxy.replay_received_message_on @expectation, &@block
-          @expectation.expected_messages_received? && @expectation.ensure_expected_ordering_received!
+          if @expectation.ordered?
+            @expectation.ensure_expected_ordering_received! && @expectation.expected_messages_received?
```

Comment:
Why do these flip around when ordered, why are we even checking the ordering when an expectation isn't ordered!